compiler/lib/linker.ml

```diff
@@ -396,20 +416,44 @@ let load_fragment ~filename f =
                 let named_values = find_named_value code in
                 Primitive.register name kind ka arity;
                 StringSet.iter Primitive.register_named_value named_values;
-                (if Hashtbl.mem provided name
-                then
-                  let _, ploc, weakdef = Hashtbl.find provided name in
-                  if not weakdef
-                  then
-                    warn
-                      "warning: overriding primitive %S\n  old: %s\n  new: %s@."
-                      name
-                      (loc ploc)
-                      (loc pi));
-                Hashtbl.add provided name (id, pi, weakdef);
-                Hashtbl.add provided_rev id (name, pi);
-                check_primitive ~name pi ~code ~requires;
-                Hashtbl.add code_pieces id (code, requires)))
+                let is_updating =
+                  if not (Hashtbl.mem provided name)
+                  then true
+                  else
+                    let _, ploc, weakdef, prev_target_env = Hashtbl.find provided name in
+                    if weakdef
+                    then true
+                    else
+                      match prev_target_env, target_env with
+                      | `Isomorphic, `Browser -> Config.Flag.is_targetting_browser_env ()
+                      | `Isomorphic, `Nodejs -> Config.Flag.is_targetting_nodejs_env ()
+                      | `Nodejs, `Isomorphic | `Browser, `Isomorphic -> false
+                      | `Nodejs, `Browser | `Browser, `Nodejs ->
```

Comment:
I believe this case cannot happen, if so, we should just fail (it is a bug) instead of emitting a warning

Reply:
You will note that the code below is currently in the code base. These are just additional constraints to make it into this existing code block. Previously you requested to make this exhaustive, which I agree with, but it’s the same duplicate symbol check that was there before. Please advise :)